packages/browser-wallet/src/background/index.ts

```diff
@@ -152,11 +163,52 @@ async function migrateNetwork(network: NetworkConfiguration) {
     }
 }
 
+/*
+ * Add the given token to all accounts of the given network, that don't already have it.
+ */
+async function addTokenToAccounts(network: NetworkConfiguration, contractIndex: number, token: TokenStorage) {
+    const tokens = await storedTokens.get(network.genesisHash);
+    const creds = await storedCredentials.get(network.genesisHash);
+    if (creds && creds.length > 0) {
+        const accountAddresses = creds.map((cred) => cred.address);
+        const newTokens = accountAddresses.reduce((accumulator, address) => {
+            const accTokens = tokens?.[address] || {};
+            if (contractIndex in accTokens) {
+                accumulator[address] = accTokens;
+            } else {
+                accumulator[address] = { [contractIndex]: [token], ...accTokens };
+            }
+            return accumulator;
+        }, {} as Record<string, Record<string, TokenStorage[]>>);
+        await storedTokens.set(network.genesisHash, newTokens);
+    }
+}
+
+/**
+ * Add euroe tokens to all accounts that don't already have it, if the migration have no already been run, and sets the euroe migration check.
+ */
+async function euroeMigration() {
+    const migrations = (await storedMigrations.get()) || [];
+    if (!migrations.includes(Migrations.euroe)) {
+        // Add metadata, if not present
+        const metadata = (await storedTokenMetadata.get()) || {};
+        const { metadataLink } = euroeTokenStorage;
+        if (!(metadataLink in metadata)) {
+            metadata[metadataLink] = EUROE_METADATA;
+            await storedTokenMetadata.set(metadata);
+        }
+        await addTokenToAccounts(mainnet, EUROE_MAINNET_INDEX, euroeTokenStorage);
+        await addTokenToAccounts(testnet, EUROE_TESTNET_INDEX, euroeTokenStorage);
+        await storedMigrations.set([...migrations, Migrations.euroe]);
+    }
+}
+
 const startupHandler = async () => {
     const network = await storedCurrentNetwork.get();
     if (network) {
         await migrateNetwork(network);
         await startMonitoringPendingStatus(network);
+        await euroeMigration();
```

Comment:
Run migrations before starting the monitoring. We could make a `migrations()` function and then move the migrations there so it is easy to add to in the future.